src/cloudwatch/modules/dimensionplugins/generic_dimensions.py

```diff
@@ -0,0 +1,53 @@
+"""
+This is the file containing generic dimension plugin classes
+"""
+
+import os
+from . import DimensionPlugin
+
+"""
+This InstanceId Dimension coming from configured host instance
+"""
+
+def dimension_get_instance_id(dimension, args):
+    dimension[args['name']] = args['value']
+
+class Dimension_InstanceId(DimensionPlugin):
+    def register_plugin(self):
+        self.func = dimension_get_instance_id
+        self.args = {
+            'name': "InstanceId",
+            'value': self.config.host
+        }
+
+
+"""
+This PluginInstance Dimension coming from collectd value plugin instance
+"""
+
+def dimension_get_plugin_instance(dimension, args):
+    dimension[args['name']] = args['value']
+
+class Dimension_PluginInstance(DimensionPlugin):
+    def register_plugin(self):
+        self.func = dimension_get_plugin_instance
+        self.args = {
+            'name': "PluginInstance",
+            'value': self.vl.plugin_instance
```

Comment:
if self.vl.plugin_instance is none, you can use "NONE". Refer to https://github.com/awslabs/collectd-cloudwatch/blob/master/src/cloudwatch/modules/metricdata.py#L139